helita/sim/ebysus.py

```diff
@@ -490,6 +569,62 @@ def get_varTime(self, var, snap=None, iix=None, iiy=None, iiz=None,
 ###########
 
 
+def write_mfr(rootname,inputdata,mf_ispecies,mf_ilevel):
+    directory = '%s.io/mf_%02i_%02i/mfr' % (rootname,mf_ispecies,mf_ilevel)
+    nx, ny, nz = inputdata.shape
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    data = np.memmap(directory+'/%s_mfr_%02i_%02i.snap' % (rootname,mf_ispecies,mf_ilevel), dtype='float32', mode='w+', order='f',shape=(nx,ny,nz,1))
+    data[:,:,:,0] = inputdata
+    data.flush()
+
+def write_mfp(rootname,inputdatax,inputdatay,inputdataz,mf_ispecies,mf_ilevel):
+    directory = '%s.io/mf_%02i_%02i/mfp' % (rootname,mf_ispecies,mf_ilevel)
+    nx, ny, nz = inputdatax.shape
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    data = np.memmap(directory+'/%s_mfp_%02i_%02i.snap' % (rootname,mf_ispecies,mf_ilevel), dtype='float32', mode='w+', order='f',shape=(nx,ny,nz,3))
+    data[:,:,:,0] = inputdatax
```

Comment:
In these cases, it is better to use `data[..., 0]` instead of `data[:,:,:,0]` as it is cleaner and more general.

Reply:
done